packages/@storefront/flux-capacitor/src/core/requests/utils.ts

```diff
@@ -59,10 +59,14 @@ namespace RequestHelpers {
       query: Selectors.pastPurchaseQuery(state),
       refinements: Selectors.pastPurchaseSelectedRefinements(state),
       skip: Selectors.pastPurchasePageSize(state) * (Selectors.pastPurchasePage(state) - 1),
-      sort: RequestAdapter.extractSort(Selectors.pastPurchaseSortSelected(state)),
     };
 
-    return <Request>{ ...request, ...overrideRequest };
+    const sort = RequestAdapter.extractPastPurchaseSort(
+      <any>overrideRequest.sort || Selectors.pastPurchaseSortSelected(state),
```

Comment:
Why is this being cast to type of any?

Reply:
In this case, the `sort` property is cast to `any` because the `overrideRequest` partially implements the `Request` interface (exposed by the `groupby-api` package). This interface says that the `sort` property can be either a 'field' sort, or a 'by id' sort. However, StoreFront expects the sort to be of the 'field'-type, which has a different shape.

See below for the interface definition:
https://github.com/groupby/api-javascript/blob/develop/src/models/request.ts#L13

Reply:
As per our IRL discussion 2018/12/27: I think that an update to what StoreFront expects in this case is required, however that update is outside of the scope of this ticket. Potentially we should resolve this down the line in another ticket. Also as a note: you mentioned that a potential implementation for this already exists on a branch, we should pull that in when we end up tackling this issue.

Reply:
This update has been described in SF-1244. Resolving.